jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/content/ContentSourceCompletableFuture.java

```diff
@@ -0,0 +1,91 @@
+//
+// ========================================================================
+// Copyright (c) 1995 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.io.content;
+
+import java.io.EOFException;
+import java.util.concurrent.CompletableFuture;
+
+import org.eclipse.jetty.io.Content;
+
+/**
+ *
+ */
+public abstract class ContentSourceCompletableFuture<X> extends CompletableFuture<X>
```

Comment:
I think this class should be either package private or moved to internal.

Reply:
It is in jetty-io and used in jetty-http.  I think it is a genuinely useful utility class, along with the other adaptor classes we have for Content.Source in jetty-io.